frontend/app/page.tsx

```diff
@@ -23,16 +23,25 @@ const theme = createTheme({
   },
 });
 
+/* check that string contains number, upper case, lower case, special character,
+and at least 10 characters */
+const validPasswordPattern = new RegExp(
+  "^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[^\da-zA-Z]).{10,}$"
+);
+
 export default function SignUp() {
   const [showPassword, setShowPassword] = useState(false);
+  const [isStrong, setIsStrong] = useState(false);
+  const [isValid, setIsValid] = useState(false);
```

Comment:
**suggestion**: Can we call these `isStrongPassword` and `isValidForm`?

Reply:
yes